src/main/java/epam/gym/dao/hibernateImpl/TraineeDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+package epam.gym.dao.hibernateImpl;
+
+import epam.gym.dao.TraineeDao;
+import epam.gym.entity.Trainee;
+import epam.gym.entity.Trainer;
+import epam.gym.entity.Training;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Repository
+public class TraineeDaoImpl implements TraineeDao {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    @Transactional
+    public Trainee create(Trainee trainee) {
+        entityManager.persist(trainee);
+        return trainee;
+    }
+
+    @Override
+    @Transactional
+    public Trainee update(Trainee trainee) {
+        return entityManager.merge(trainee);
+    }
+
+    @Override
+    @Transactional
+    public Optional<Trainee> findById(Long id) {
+        return Optional.ofNullable(entityManager.find(Trainee.class, id));
+    }
+
+    @Override
+    @Transactional
+    public void deleteById(Long id) {
+        Trainee trainee = entityManager.find(Trainee.class, id);
+        if (trainee != null) {
+            entityManager.remove(trainee);
+        }
+    }
+
+    @Override
+    public Set<String> getExistingUsernames() {
+        List<String> usernames = entityManager.createQuery("SELECT username FROM Trainee", String.class).getResultList();
+        return new HashSet<>(usernames);
+    }
+
+    @Override
+    @Transactional
+    public Optional<Trainee> selectTraineeByUsername(String username) {
```

Comment:
I'd follow the single format of method names `findTraineeByUsername`

Reply:
fixed